components/pkg-export-kubernetes/src/bind.rs

```
@@ -0,0 +1,56 @@
+// Copyright (c) 2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::ArgMatches;
+use std::str::FromStr;
+
+use error::Error;
+
+#[derive(Debug, Clone)]
+pub struct Bind {
+    pub name: String,
+    pub service: String,
+    pub group: String,
+}
+
+impl FromStr for Bind {
+    type Err = Error;
+
+    fn from_str(bind_spec: &str) -> Result<Self, Error> {
+        let split: Vec<&str> = bind_spec.split(":").collect();
+        if split.len() < 3 {
```

If we specify a bind spec like "foo:bar:baz:quux" then we silently drop the quux part. So I'd either put a stricter check here I think - `split.len() != 3` or use `bind_spec.splitn(3, ":")`, so user can specify a group name with colons if they need it.